src/hotspot/share/runtime/continuation.cpp

```diff
@@ -2077,6 +2071,7 @@ static inline int prepare_thaw0(JavaThread* thread, bool return_barrier) {
   stackChunkOop chunk = jdk_internal_vm_Continuation::tail(continuation);
   assert(chunk != nullptr, "");
 
+  // Comment needed: Why would the tail chunk be empty? Why do you get the parent?
```

Comment:
Here, can you explain what's happening in this function?

Reply:
Then we'll do more functions :)

Reply:
We make room on the stack to thaw frames. As we'll either thaw the topmost non-empty chunk or one frame, we just need to make room for the topmost non-empty chunk.

Reply:
So here the topmost chunk may be null.  Why is it null?  Is it null because it was already partially thawed and you leave space for a new chunk in the chunk memory?  In this case, you continue to the parent (next) frame.  Did I get that right?

Reply:
Not null, but empty. I can't remember at the moment why we do it like that. I'll take a closer look.